### ISSUE TITLE
Harden Rust-first architecture guardrails and make mirrors explicit, costly exceptions

### ISSUE THREAD

**Issue by rhythmatician** [rhythmatician/FractalSync#119]
Follow-up to ADR 0001 and the architecture guardrail work from #90. Triggered by the #107 / PR #117 leak where production `modelInference.ts` reimplemented `JuliaViewState` transition semantics in JavaScript despite Rust being the declared authority.

## Problem

Our documented architecture is stronger than our machine backpressure.

ADR 0001 says shared/domain behavior belongs in `runtime-core` and other languages consume bindings, but the current guardrail is mostly a targeted detector for already-known formulas/concepts. That allowed a new class of domain behavior to leak into TypeScript simply because its math looked generic: clamps, exponential zoom integration, angle wrapping, color-state mutation, harmony thresholds, etc.

The mirror manifest also makes mirrors too cheap. A developer/agent can currently turn a domain reimplementation into an allowed exception by adding a file-level mirror entry plus parity tests. That creates exactly the maintenance burden we are trying to avoid: duplicate authority, change amplification, and annoying parity suites.

We want the incentives reversed:

> **The easiest path must be to add/extend the Rust API and bindings. Creating a mirror must be harder, explicitly justified, narrowly scoped, parity-pinned, and planned for deletion.**

## Canonical architecture invariant

Unless a genuine platform capability prevents it:

> **Rust owns every deterministic domain state transition and mathematical/domain transformation used by runtime or training.**

Language responsibilities:

### TypeScript
Allowed to own:
- browser APIs;
- UI;
- rendering/WebGL integration;
- WASM binding adapters;
- ONNX/browser transport and metadata IO.

TypeScript must **not** implement product/domain semantics merely because a WASM binding is missing.

### Python
Allowed to own:
- training/optimization;
- autograd/backward surrogates;
- experiment orchestration;
- offline diagnostics/tooling.

Python must **not** become a second forward implementation of feature semantics, cadence, observation packing, model IO ordering, Map geometry, Physics, Controls, persistent view-state transitions, or other shared domain behavior merely for convenience.

### Training exception

The preferred differentiable pattern is:

```text
Rust forward semantics
    +
declared training-only surrogate backward gradient
```

A Python forward reimplementation requires a much stronger capability justification than "PyTorch needs gradients" if the Rust forward can still be called and an STE/custom backward supplied.

## Mirror admission rule

Mirrors are **forbidden by default**.

A mirror may be admitted only when a concrete platform capability makes calling/extending the Rust authority impossible or materially incompatible with the required operation.

The following are **not valid capability gaps**:

- "the binding does not expose it yet";
- "it is faster/easier to write in JS/Python";
- "the browser needs the value";
- "the trainer needs the value" without demonstrating why Rust-forward + custom backward is insufficient;
- "temporary" implementation convenience;
- avoiding Rust/WASM/PyO3 work.

**If a binding is missing, stop and add the Rust operation/binding first. Do not create a mirror to unblock implementation.**

## Strengthen `architecture_mirrors.json`

Replace broad file-level permission with narrowly scoped exception records. Each behavioral/differentiable mirror should declare at least:

```json
{
  "path": "...",
  "kind": "differentiable_mirror",
  "rust_authority": "...",
  "mirrored_symbols": ["exact_symbol_a", "exact_symbol_b"],
  "capability_gap": "concrete platform limitation",
  "why_binding_cannot_solve_it": "...",
  "alternatives_considered": [
    "extend Rust binding",
    "move operation entirely into Rust"
  ],
  "tracking_issue": "#...",
  "sunset_condition": "...",
  "parity": ["real executable parity test"]
}
```

Requirements:

- every behavioral/differentiable mirror names **exact mirrored symbols/operations**, not permission for arbitrary domain logic anywhere in the file;
- every behavioral/differentiable mirror has a tracking issue;
- every behavioral/differentiable mirror has a sunset condition, not only `transitional_mirror`;
- every mirror addition/expansion lands with its required parity test(s) in the same change;
- parity entries must resolve to real executable tests/gates;
- changing `architecture_mirrors.json` should be treated as an architecture exception, not ordinary implementation work;
- diagnostics/experiments that are provably not consumed by runtime/training/model behavior should be represented separately from behavioral mirrors so they do not normalize duplication.

## Structural guardrail, not only formula regexes

Keep targeted formula rules where useful, but add structural enforcement so novel semantics cannot evade CI just because we have not seen their formula before.

The guardrail should enforce the invariant that canonical domain state transitions originate in Rust and non-Rust surfaces are adapters/consumers unless an exact manifested exception permits otherwise.

Candidate enforcement mechanisms to implement/test:

1. `shared/canonical_surfaces.json` identifies the Rust transition/API for each canonical subsystem, not merely `authority: rust`.
2. Known Rust-owned domain state types are opaque to production JS/Python business logic: they may be constructed/passed/read for rendering/training where necessary, but deterministic transition rules must call bindings rather than mutate/rederive semantics.
3. Manifested mirror permission is symbol/operation-scoped rather than whole-file scoped.
4. CI fails if non-Rust code implements a canonical transition without an admitted exception.
5. Cheap smoke alarms flag production comments/identifiers such as `JS mirror`, `Python mirror`, `simplified mirror`, `lightweight mirror`, or `for full authority, wire...`; these are not sufficient enforcement alone, but should force explicit review because such phrases should almost never appear on production paths.
6. Guardrail tests include negative fixtures demonstrating that generic-looking formulas/state mutations (clamp/wrap/threshold/integration) cannot bypass the rule when they implement a known Rust-owned transition.
7. The failure message should instruct the developer to **add/extend the Rust binding**, not suggest adding a mirror manifest entry as the default fix.

## Immediate leak to repair: PR #117 / #107

`frontend/src/lib/modelInference.ts` currently contains a JavaScript mirror of Rust-owned `JuliaViewState` semantics, including:

- zoom integration/clamping;
- rotation wrapping;
- hue/chroma/lightness/accent integration/clamping;
- harmony threshold/state transition.

Replace that with a Rust-owned transition API, conceptually:

```text
JuliaViewState::step(JuliaViewControls)
    -> JuliaViewState
```

Expose it through WASM and have the browser call it. Delete the JS formulas/state-transition implementation. Add cross-surface parity/determinism coverage as appropriate.

The browser should be orchestration only:

```text
view_state = wasm.step_julia_view(view_state, view_controls)
```

## Audit the existing mirror inventory

Re-review every current `architecture_mirrors.json` entry under the stricter admission standard.

For each entry, decide one of:

1. **Delete mirror / extend Rust API or binding** — preferred.
2. **Narrow to exact symbols** and retain with concrete capability justification + parity + sunset.
3. **Reclassify as diagnostic/experimental** only if it is mechanically proven not to participate in runtime/training/model behavior.

Pay particular attention to:

- `backend/src/cspace_proxies.py` — differentiable surrogates are likely legitimate only where autograd requires them; prefer Rust forward oracle + surrogate backward and narrow permission to exact operations.
- `backend/src/python_feature_extractor.py` — current justification includes dataset normalization / Rust API convenience; determine whether extending the Rust API would eliminate this mirror.
- `frontend/src/lib/__tests__/orbitSynthesizer.mock.ts` — test-only behavior; decide whether a behavioral mirror is truly necessary or whether generated fixtures/fakes can reduce parity burden.
- `backend/src/visual_metrics.py` and `backend/src/julia_gpu.py` — confirm whether rendering/training capability genuinely requires these mirrors and narrow scope accordingly.
- `frontend/src/lib/modelInference.ts` — remove domain semantics; transport-only destination.
- diagnostic/experimental entries — separate from runtime/training mirror exceptions where possible.

Do not grandfather an existing mirror merely because parity tests already exist. The question is whether the duplicate implementation is still worth owning.

## ADR 0001 amendment

Make the policy explicit in ADR 0001:

> **Do not create a mirror to unblock implementation. Stop, extend the Rust API/bindings, then continue. A mirror is an architectural exception, not a development technique.**

Also state that missing bindings are not a capability justification and that all admitted behavioral/differentiable mirrors incur parity + tracking + sunset obligations.

## Acceptance

- [ ] ADR 0001 explicitly states the Rust-first language boundary and mirror admission policy above.
- [ ] Production TypeScript contains no `JuliaViewState` domain transition implementation from PR #117; it calls Rust/WASM authority instead.
- [ ] Rust exposes the needed Julia view-state transition API through WASM; deterministic browser behavior is tested.
- [ ] `architecture_mirrors.json` requires exact mirrored symbols/operations rather than granting whole-file semantic permission.
- [ ] Behavioral/differentiable mirrors require `capability_gap`, `why_binding_cannot_solve_it`, `alternatives_considered`, `tracking_issue`, `sunset_condition`, and real parity checks.
- [ ] "Missing binding" / convenience cannot satisfy mirror admission validation.
- [ ] Every existing mirror is audited and either removed, narrowed/justified, or reclassified as truly diagnostic/experimental.
- [ ] Training mirrors prefer Rust-forward + surrogate-backward wherever technically possible.
- [ ] Guardrail failure messages direct developers toward extending Rust/bindings first.
- [ ] CI includes negative tests for novel/generic-looking domain state-transition leakage, not only known formula signatures.
- [ ] CI includes a cheap smoke alarm for production `* mirror` / `for full authority...` implementation markers.
- [ ] Adding/expanding a behavioral mirror without the complete justification/parity/sunset contract fails CI.
- [ ] `canonical_surfaces.json` (or equivalent machine-readable architecture data) identifies canonical Rust transition APIs sufficiently for the guardrail to reason about ownership.
- [ ] Frontend remains browser/UI/rendering/transport; Python remains training/autograd/orchestration/diagnostics; deterministic shared domain semantics live in Rust by default.

## Non-goals

- Eliminating Python training or TypeScript browser code.
- Forcing browser-specific APIs, UI, rendering, or ONNX transport into Rust.
- Forcing optimizer/autograd machinery into Rust.
- Prohibiting diagnostic/experimental code that is mechanically isolated from production/training/model semantics.
- Maintaining parity tests for duplicate behavior that can instead be deleted by extending the Rust API.
